vortex-array/src/compute/filter.rs

```diff
@@ -234,6 +228,20 @@ impl TryFrom<&dyn Array> for Mask {
     }
 }
 
+pub fn arrow_filter_fn(array: &dyn Array, mask: &Mask) -> VortexResult<ArrayRef> {
+    let values = match &mask {
```

Comment:
I don't think you need this bit here since it's handled above based on true_count?

Reply:
unreachable?